project/platform/src/net/ashald/envfile/platform/ui/EnvFileConfigurationEditor.java

```diff
@@ -108,10 +109,16 @@ public static void writeExternal(@NotNull RunConfigurationBase configuration, @N
     }
 
     public static void patchCommandLine(@NotNull RunConfigurationBase configuration, @Nullable RunnerSettings runnerSettings, @NotNull GeneralCommandLine cmdLine) throws ExecutionException {
-        EnvFileSettings state = configuration.getUserData(USER_DATA_KEY);
-        if (state != null && state.isEnabled()) {
-            Map<String, String> source = cmdLine.getEnvironment();
+        patchEnvironmentVariables(configuration, cmdLine.getEnvironment());
+    }
 
+    public static void patchJavaParameters(@NotNull RunConfigurationBase configuration, JavaParameters params, RunnerSettings runnerSettings) throws ExecutionException {
```

Comment:
Done !
Just for information, I couldn't test it with RubyMine and PyCharm but it should work :)